articles/3_sass-selectors.md

```diff
@@ -0,0 +1,237 @@
+
+## A note about SASS vs SCSS
+A lot of the time you will see SASS files with the file extension `.scss`. This denotes a file with *Sassy CSS* syntax. The difference being that any valid `.css` file can be read by an SCSS compiler and it will produce itself. There are some additional differences between SASS and SCSS, the largest of which being that SASS does not require curly braces, instead using tabs (similar to Python).
+
+For this post (and all following, unless I specify otherwise), I will be using SCSS syntax. If you're interested in learning more about the pro's and con's of each syntax, read this article [here](http://thesassway.com/editorial/sass-vs-scss-which-syntax-is-better).
+
+## Nesting
+
+One of the most annoying parts about writing vanilla CSS is that it's very hard to keep organized without tons of documentation, especially in a large system. A big part of why this is messy, is that its hard to tell which elements are children of which parents, so elements will inherit unexpected styling thats difficult to debug.
+
+In SASS, you can nest selectors like this:
+
+```scss
+.parent {
+  /* styling goes here */
+
+  .descendant {
+    /* descendant-specific styles go here */
+  }
+}
+```
+
+And it will get compiled into CSS that looks like this:
+
+```css
+.parent {
+  /* styling goes here */
+}
+
+.parent .descendant {
+  /* descendant-specific styles go here */
+}
+```
+
+Note that I'm using `descendant` instead of `child`, because this nesting works for any descendant of the parent, not just its direct children.
+
+For added specificity, you can also use this nesting with any parent and any descendant selectors (see [A Crash Course in CSS Selectors: Part 1](1_selectors.md) if you need a refresher).
+
+
+## The Ampersand
+
+If you want to get more ambitious than just nesting children from parents, you can use an ampersand (`&`) as a concatenator within nested SASS structures. There are some other idiosyncrasies with the `&`, so we will go into each of them with more detail.
+
+![Code Compilation](../images/scss-to-css.gif)
+
+*Image via css-tricks.com*
+
+As a simple example, let's revisit the above example and how we could use an ampersand to accomplish the same task.
+
+```scss
+.parent {
+  & .descendant {
+    /* styles */
+  }
+}
+```
+
+Is the exact same as
+
+```scss
+.parent {
+  .descendant {
+    /* styles */
+  }
+}
+```
+
+Because they both compile to
+
+```css
+.parent .descendant {
+  /* styles */
+}
+```
+
+Note that this is not the preferred way to nest child selectors (I prefer the first method outlined above, as it is much cleaner). This is just an example of how the ampersand can be used. This method is not wrong, its just unnecessary when a simpler way is also available.
```

Comment:
I'd use an example of `&:hover`, etc. since that is the most common use for `&`.